README.md

```diff
@@ -1,707 +1,1799 @@
+
+---
+
+## Table of Contents
+
+- [Installation](#installation)
+  - [Redux Toolkit](#redux-toolkit)
+  - [Standalone](#standalone)
+- [Basic Usage](#basic-usage)
+- [Terminology](#terminology)
+- [How Does `Reselect` Work?](#how-does-reselect-work)
+  - [Cascading Memoization](#cascading-memoization)
+  - [`Reselect` Vs Standard Memoization](#reselect-vs-standard-memoization)
+  - [Why `Reselect` Is Often Used With `Redux`](#why-reselect-is-often-used-with-redux)
+- [API](#api)
+  - [**`createSelector`**][`createSelector`]
+  - [**`defaultMemoize`**][`defaultMemoize`]
+  - [**`weakMapMemoize`**][`weakMapMemoize`]
+  - [**`autotrackMemoize`**][`autotrackMemoize`]
+  - [**`createSelectorCreator`**][`createSelectorCreator`]
+  - [**`createStructuredSelector`**][`createStructuredSelector`]
+  - [**`createCurriedSelector`**][`createCurriedSelector`]
+- [Debugging Tools](#debuggingtools)
+- [What's New in 5.0.0?](#v5summary)
+- [Optimizing `Reselect`](#optimizing-reselect)
+- [FAQ](#faq)
+  - [Why isn’t my selector recomputing when the input state changes?](#why-isnt-my-selector-recomputing-when-the-input-state-changes)
+  - [Why is my selector recomputing when the input state stays the same?](#why-is-my-selector-recomputing-when-the-input-state-stays-the-same)
+  - [Can I use `Reselect` without `Redux`?](#can-i-use-reselect-without-redux)
+  - [How do I create a selector that takes an argument?](#how-do-i-create-a-selector-that-takes-an-argument)
+  - [The default memoization function is no good, can I use a different one?](#the-default-memoization-function-is-no-good-can-i-use-a-different-one)
+  - [How do I test a selector?](#how-do-i-test-a-selector)
+  - [Can I share a selector across multiple component instances?](#can-i-share-a-selector-across-multiple-component-instances)
+  - [Are there TypeScript Typings?](#are-there-typescript-typings)
+  - [I am seeing a TypeScript error: `Type instantiation is excessively deep and possibly infinite`](#i-am-seeing-a-typescript-error-type-instantiation-is-excessively-deep-and-possibly-infinite)
+  - [How can I make a curried selector?](#how-can-i-make-a-curried-selector)
+  - [How can I make pre-typed version of `createSelector` for my root state?](#how-can-i-make-pre-typed-version-of-createselector-for-my-root-state)
+  - [What if I want to use `createSelector` without memoization?](#what-if-i-want-to-use-createselector-without-memoization)
+- [External References](#external-references)
+- [Related Projects](#related-projects)
+- [License](#license)
+- [Prior Art and Inspiration](#prior-art-and-inspiration)
+
+---
+
+## Terminology
+
+<details><summary><b>Click to expand</b></summary>
+
+- <a name="selector-function"></a>[**`Selector Function`**](#selector-function): A function that takes the `Redux` store state (or a part of it) as an argument and returns data derived from that state.
```

Comment:
:hand: I think it's worth keeping the "not actually required to use Redux" aspect here.  How about:

```suggestion
- <a name="selector-function"></a>[**`Selector Function`**](#selector-function): A function that accepts one or more JS values as input arguments, and derives a result.  When used with Redux, the first argument is typically the entire Redux store state.
```